### PR TITLE
Fix coupon definition

### PR DESCRIPTION
**Pull request by i-yasko**
## Summary
Fix coupon discriminator definition, add nullable to expiredTime property

## Checklist

- [x] Writing style
- [x] API design standards

### CODE REVIEW

**Review by ashkarpetin — DISMISSED** [review]

File: openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml [hunk 1]
```diff
@@ -1,51 +1,53 @@
...
-              minimum: 1
-              description: Total quantity of the associated plan ID that is received.
+required:
+  - type
+  - buy
+  - get
+properties:
+  type:
+    type: string
+    description: Coupon restriction type.
```
Comment (noon-dawg) [inline openapi/components/schemas/Coupon/CouponRestrictions/restrict-to-bxgy.yaml:14]: nitpick: 3 nouns in a row 
```suggestion
    description: Type of coupon restriction.
```

File: openapi/components/schemas/Coupon/CouponRestrictions/total-redemptions.yaml [hunk 2]
```diff
@@ -1,10 +1,14 @@
...
+  - type
+  - quantity
+properties:
+  type:
+    type: string
+    description: Coupon restriction type.
+    enum: ["total-redemptions"]
+  quantity:
+    type: integer
+    description: Total redemptions quantity.
```
Comment (noon-dawg) [inline openapi/components/schemas/Coupon/CouponRestrictions/total-redemptions.yaml:13]: suggestion: I feel that 'permitted' helps convey the meaning
```suggestion
    description: Permitted total number of redemptions.
```

File: openapi/components/schemas/Coupon/CouponRestrictions/minimum-order-amount.yaml [hunk 3]
```diff
@@ -1,18 +1,19 @@
...
+  - amount
+  - currency
+properties:
+  type:
+    type: string
+    description: Coupon restriction type.
+    enum: ["minimum-order-amount"]
+  amount:
+    type: integer
+    description: Minimum order quantity.
```
Comment (noon-dawg) [inline openapi/components/schemas/Coupon/CouponRestrictions/minimum-order-amount.yaml:17]: suggestion: I think this needs more detail. I'm not sure if my suggestion is correct
```suggestion
    description: Minimum order quantity that is required to apply the coupon.
```

**Review by noon-dawg — COMMENTED** [review]
All minor nitpicks and suggestions!
Let me know if you have any questions

**Review by noon-dawg — APPROVED** [review]

**Review by ashkarpetin — APPROVED** [review]

**Pull request merged**